### PR TITLE
Adding a check for the fund adjustment

### LINKED ISSUE #211: Validate the transaction numbers in the peoplesoft Fund adjustment
When the file is retrieved from people soft we need to validate that no duplicates transaction ids have been sent from the system in the past.

Send an email to Mark, Kevin, Carolyn and do not transfer the file to Alma.  Put the file in an error directory.

### PR DESCRIPTION
**Pull request by carolyncole**
We do not want to process a transaction multiple times

fixes #211

### CODE REVIEW

**Comment by carolyncole** [comment]
@mzelesky Does this look correct?

**Comment by mzelesky** [comment]
So is this tracking invoices we submit and fund adjustments we receive?

I'll leave a few other comments in a review.

File: app/mailers/invoice_error_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+class InvoiceErrorMailer < ApplicationMailer
+  def report(duplicate_ids:)
+    @duplicate_ids = duplicate_ids
+    mail(to: "cac9@princeton.edu, mzelesky@princeton.edu, kr2@princeton.edu", subject: 'PeopleSoft Duplicate Invoices')
```
Comment (mzelesky) [inline app/mailers/invoice_error_mailer.rb:5]: The invoices are generated by Alma and sent to PeopleSoft; they are not sent from PeopleSoft. So this should have a subject of 'Alma Duplicate Invoices.'

File: app/models/alma_fund_adjustment/adjustment_check.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      status
+    end
+
+    private
+
+    def process_file(file)
+      status = true
+      data = ::CSV.read(file, headers: true)
+      ids = data.map { |row| row['TRANSACTION_REFERENCE_NUMBER'] }
+      already_processed = ids.select { |id| PeoplesoftInvoice.where(invoice_id: id).count.positive? }
```
Comment (mzelesky) [inline app/models/alma_fund_adjustment/adjustment_check.rb:28]: These are not invoices, they are fund transactions.

**Review by mzelesky — CHANGES_REQUESTED** [review]
In general, this looks to me like fund adjustments are being lumped in with invoices. If an invoice is generated from Alma with the same invoice ID, that doesn't mean we would want to skip that invoice. In fact, if an invoice fails one of the other checks we have in place, we would expect operators in Alma to correct the invoice and re-submit it.

The only duplicates I'm concerned about are duplicates from the fund adjustment files.

File: app/mailers/invoice_error_mailer.rb (hunk 1 above)
Comment (carolyncole) [reply app/mailers/invoice_error_mailer.rb:5]: @mzelesky maybe I have this incorrect, but the data is coming from PeopleSoft that we are checking (Correct?). I am looking at Duplicate data from the PeopleSof Report.  The duplicate data is not from Alma, or is Alma creating a Duplicate?
Should this really say `PeopleSoft Duplicate Transaction`?

**Comment by carolyncole** [comment]
@mzelesky A ton of the code was written prior to you correcting the ticket from Voucher Feed to Fund Adjustment.  I went ahead and renamed some of the classes to match the Fund Adjustment process better.

**Comment by carolyncole** [comment]
@mzelesky I am not tracking Invoices at all, that was just based on the information I saw in the Voucher Feed.  There was a disconnect on my part of what people soft feed we were checking, and no one corrected the ticket after I wrote it so I assumed it was correct.

**Comment by mzelesky** [comment]
I apologize for it taking a few hours for me to review the ticket after it was initially created.

**Comment by carolyncole** [comment]
@mzelesky no problem.  I know you are crazy busy.  I was just explaining where the strange code came from...

**Comment by mzelesky** [comment]
This looks good now.

File: app/mailers/invoice_error_mailer.rb (hunk 1 above)
Comment (mzelesky) [reply app/mailers/invoice_error_mailer.rb]: Yes. `PeopleSoft Duplicate Fund Transactions`.

**Review by mzelesky — APPROVED** [review]
This looks good.

**Review by kevinreiss — APPROVED** [review]

**Pull request merged**